### ISSUE TITLE
false positive redefined var

### ISSUE THREAD

**Issue by borkdude** [borkdude/clj-kondo#244]
take into account conditionals:
```
(if (and (exists? js/Symbol)
         (identical? (goog/typeOf js/Symbol) "function"))
  (def ITER_SYMBOL (.-iterator js/Symbol))
  (def ITER_SYMBOL "@@iterator"))
```

`declare` is a no-op when the var already exists and this should not lead to a warning:
```
cljs.user=> (def foo 1)
#'cljs.user/foo
cljs.user=> (declare foo)
#'cljs.user/foo
cljs.user=> foo
1
```
